File: backend/src/main/java/com/banking/banking/model/Account.java
package com.banking.banking.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.UuidGenerator;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "accounts")
@Data
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class Account {
    @Id
    @GeneratedValue
    @UuidGenerator
    private UUID id;
    @Column(nullable = false, unique = true)
    private String number;
    @Column(nullable = false, unique = true)
    private String name;
    @Column(nullable = false)
    private BigDecimal balance = BigDecimal.ZERO;
    @CreatedDate
    private LocalDateTime createdAt;
    @LastModifiedDate
    private LocalDateTime updatedAt;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false, columnDefinition = "foreign key(user_id) references users(id) on delete cascade")
    private User user;
}

File: frontend/src/api/models/account.ts
export interface AccountDto {
  id: string;
  name: string;
  number: string;
  balance: number;
  createdAt: string;
}

export interface AccountSearchRequest {
  name: string;
  number: string;
}

export interface TransferRequest {
  from: string;
  to: string;
  amount: number;
}

export interface AccountCreateRequest {
  balance: number;
  name: string;
}

export interface AccountUpdateRequest {
  name: string;
}
